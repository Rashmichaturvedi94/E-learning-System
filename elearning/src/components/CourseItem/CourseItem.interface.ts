export interface CourseItemProps {
  // TODO: add required props below
  item?: any;
}
