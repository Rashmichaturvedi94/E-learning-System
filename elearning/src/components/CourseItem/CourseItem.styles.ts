import styled from 'styled-components/native';
import { CourseItemProps } from './CourseItem.interface';

export const CourseItem = styled.View<Pick<CourseItemProps, 'myProp'>>`
  /* Add styles here */
`;
