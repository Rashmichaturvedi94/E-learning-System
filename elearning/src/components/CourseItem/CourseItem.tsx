import React, { FC } from 'react';
import {
  CourseImage,
  CourseContainer,
  CourseTitle,
  FavIcon,
  FavContainer,
  FavCount,
} from './CourseItem.styles';
import { CourseItemProps } from './CourseItem.interface';

export const CourseItem: FC<CourseItemProps> = ({ item }) => {
  return (
    <CourseContainer>
      <CourseImage
        source={{
          uri: item.imageUrl,
        }}
      />
      <FavContainer>
        <FavCount>{item.favCount}</FavCount>
        <FavIcon name="favorite" size={24} />
      </FavContainer>
      <CourseTitle>{item.title}</CourseTitle>
    </CourseContainer>
  );
};
