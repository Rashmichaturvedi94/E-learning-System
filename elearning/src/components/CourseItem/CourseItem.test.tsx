import React from 'react';
import { render } from '@testing-library/react-native';
import { CourseItem } from './CourseItem';

describe('components/CourseItem', () => {
  it('should render', () => {
    const component = render(<CourseItem />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
