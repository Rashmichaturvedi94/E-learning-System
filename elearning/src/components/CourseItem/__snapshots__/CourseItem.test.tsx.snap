// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CourseItem should render 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 5,
        "marginBottom": 0,
        "marginLeft": 4,
        "marginRight": 4,
        "marginTop": 0,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://upload.wikimedia.org/wikipedia/commons/7/78/Swift_logo.png",
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 8,
          "height": 160,
          "width": 200,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "borderRadius": 5,
          "flexDirection": "row",
          "marginBottom": 8,
          "marginLeft": 8,
          "marginRight": 8,
          "marginTop": 8,
          "paddingBottom": 1,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 1,
          "position": "absolute",
          "right": 0,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 13,
            "marginRight": 4,
          },
        ]
      }
    >
      22
    </Text>
    <View
      style={
        Object {
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "bottom": 0,
              "color": "white",
              "height": 24,
              "left": 0,
              "right": 0,
              "top": 0,
              "width": 24,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 24,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            î¡½
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "borderBottomLeftRadius": 5,
          "borderBottomRightRadius": 5,
          "bottom": 0,
          "height": 40,
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 17,
            "fontWeight": "600",
            "marginBottom": 6,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 6,
          },
        ]
      }
    >
      Swift
    </Text>
  </View>
</View>
`;
